Update documentation + load dotenv

imenattatra opened the pull request:
## What type of PR is this? 

- [x] Refactor (because of calling `load_dotenv`)
- [x] Documentation


## Description

This change suggestion include 4 small parts : 

~1- update of the requirements-metr.txt file that was used before...
even if now it is not really needed but i had issues with poetry not installing all the packages and tests were failing because of that locally but they were working on docker image. So, i got the requirements from the docker image and put them in this file , this way, the dependencies can be installed with `pip3 install -r requirements-metr.txt` , if you prefer to remove that file i would be okay with it since it is just a temporary workaround~

1- remove the requirements-metr.txt file since it is not needed anymore and we use poetry to install packages as redash people 

2- add `.env-example` file in order to help setting up the project locally since we usually have an example that we copy and fill ... 
they have a command `make .env` that would create the `.env ` file but, not only it wouldn't be easy to notice, it also creates it without the DATABASE_URL variable , so, i thought it would be a good idea to have it

3- make sure the env variables in the `.env` file are loaded by calling `load_dotenv`...
I honestly don't know why they were not doing this already since we install the `python-dotenv` package and they introduce `.env` file 

4- update the readme file with instructions to run both backend and frontend locally without docker.

## Extra note

The pre-commit is failing but not because of our changes , so I chose not to fix line of codes that we don't touch in order to avoid a lot of conflicts with `getredash` code later .


## How is this tested?

- [x] Unit tests (pytest, jest)

imenattatra committed:
add env-example and enable dotenv

imenattatra committed:
update requirements-metr.txt

imenattatra committed:
update readme with notes about running both backend and frontend and some test notes

thiagogds reviewed (approved):
Great work!

I would prefer to not rely on the `requirements-metr` and only use poetry to be as close as possible to their development structure.

Have you tried installing with poetry using `poetry install --with dev`?

imenattatra commented:
> Great work!
> 
> I would prefer to not rely on the `requirements-metr` and only use poetry to be as close as possible to their development structure.
> 
> Have you tried installing with poetry using `poetry install --with dev`?

Oh yes makes sens, this command seems to install the dependencies needed actually, thanks a lot, i think `dev` should already be included within the `poetry install --only main,all_ds,dev` command.
Seems like with poetry my issue would only be `mysqlclient` requirement that i forgot about because I skipped it and it does not affect tests , i documented it [here](https://github.com/orgs/metr-systems/projects/18/views/1?pane=issue&itemId=73000646) . I will try to fix it and update this PR and the documentation issue :D

imenattatra committed:
update readme : use working version for mysql-client on mac

imenattatra committed:
remove requirement-metr file since it is not needed anymore/ we use poetry

imenattatra commented:
@thiagogds i am so happy , it works now with poetry now for me :D 

I updated the readme and removed the requirements-metr file since it is not needed anymore

thiagogds commented on metr_docs/README.md:
```diff
@@ -168,10 +168,13 @@ With that done, install the rest of the Python dependencies:
 (redashvenv1) $ pip3 install wheel  # "wheel" needs to be installed by itself first
 (redashvenv1) $ pip3 install --upgrade black ruff launchpadlib pip setuptools
 (redashvenv1) $ pip3 install poetry
+(redashvenv1) $ poetry install --with dev
```
The `dev` is also being installed in the line bellow. The problem is that it also installs all_ds, which is only necessary in the production environment. Those are the dependencies for the data sources.

I think we can change the readme to have something like
```
(redashvenv1) $ poetry install --with dev
# If you need to have the data sources dependencies locally, also install them with
(redashvenv1) $ poetry install --only all_ds
```

imenattatra replied:
thanks a lot , yeah i tested it and then changed it in the readme :D

imenattatra committed:
update readme file - remove not needed note

imenattatra committed:
update readme file for poetry commands      poetry install only for all_ds in case you want to install data sources

jhlund reviewed (approved):
Not really understanding all of it since my knowledge of redash is rather limited. But it looks good :+1:

saschahausmann reviewed (commented):
Thank you for including me in the review process. <3
